### ISSUE TITLE
npm plugin overwrite every scoped packages as public

### ISSUE THREAD

**Issue by SirbyAlive** [intuit/auto#858]
**Describe the bug**

Npm publish of private scoped packages override the npm setting and turn the package public. 

**To Reproduce**

Have a scoped package, release it, it will be public.
Change the npm package setting to `private` manually, release again, the package will become public again.

**Expected behavior**

Scoped packages should be published as private.

**Screenshots**

**Desktop (please complete the following information):**

irrelevant

**Additional context**

The alternative of setting `"private": true` on the package.json is not a solution since npm will refuse to publish it.
```
npm ERR! code EPRIVATE
npm ERR! This package has been marked as private
npm ERR! Remove the 'private' field from the package.json to publish it.
```

**Comment by SirbyAlive** [comment]
Looking at the tests (https://github.com/intuit/auto/blob/master/plugins/npm/__tests__/npm.test.ts#L584) I can see that this is by design.
But since the case where we have a scoped package with `"private": true` in not publishable, the next test is not applicable.

A solution would be to introduce a `.autorc` option to override the `--access` to `public` or `restricted`.

And change https://github.com/intuit/auto/blob/master/plugins/npm/src/index.ts#L648
```ts
const publishArgs = ['--tag', 'canary'];
await execPromise(
  'npm',
  !isPrivate && isScopedPackage
    ? ['publish', '--access', 'public', ...publishArgs, ...verboseArgs]
    : ['publish', ...publishArgs, ...verboseArgs]
);
```
to be closer to the expected NPM behavior https://docs.npmjs.com/creating-and-publishing-private-packages.
I guess no specific checks are actually required like so:
```typescript
let publishArgs = ['--tag', 'canary'];
if (this.access) {
  publishArgs = [...publishArgs, '--access', this.access];
}
await execPromise(
  'npm',
  ['publish', ...publishArgs, ...verboseArgs]
);
```

**Comment by hipstersmoothie** [comment]
I agree this is not good. This code was put in to make it easier for publishing scoped packages for the first time. But since the offending code is only for single package this doesn't make too much sense. Matching the npm behavior is probably desired.

https://github.com/intuit/auto/pull/868

**Comment by adierkens** [comment]
:rocket: Issue was released in v9.0.0 :rocket:
